### PR TITLE
Travis CI: Test on Python 2.7 and 3.8

### PR DESCRIPTION
**Pull request by cclauss**
This PR streamlines the Travis CI processing by bringing all functionality into `.travis.yml` and removing the separate scripts.  It also modernizes the Python versions.  The tests pass on Linux but [fail on macOS](https://travis-ci.com/cclauss/pyftpdlib/jobs/291703563#L457) but I can not determine why.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+          language: generic
+          env: PYTHON=python2.7
+        - os: osx
+          osx_image: xcode11.3
+          language: generic
+          env: PYTHON=python3.7
 install:
-    - ./.ci/travis/install.sh
+    - $PYTHON -m pip install --upgrade pip setuptools
+    - $PYTHON -m pip install --upgrade coveralls flake8 mock psutil pyopenssl pysendfile pytest six
```
Comment (lurch) [inline .travis.yml]: Does it make more sense to store these dependencies in e.g. a `test-requirements.txt` file?

**Comment by lurch** [comment]
Why 3.8 on Linux but 3.7 on OSX?

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: There are two philosophies...
1) Keep all that we need in a single file.  _or_
2) Spread state and commands across multiple files.

This PR was focused on 1) by moving state and commands into a single file and deleting three other files.

Do contributors run these same test locally?

**Comment by cclauss** [comment]
Like https://en.wikipedia.org/wiki/Crop_rotation, it is good to run on multiple versions of Python to catch subtle differences between releases.  Besides, using pyenv to install Python 3.8 on macOS takes significant time/cpu/energy.

File: .travis.yml (hunk 1 above)
Comment (lurch) [reply .travis.yml]: It was just a query, I'll leave the decision up to @giampaolo

**Comment by lurch** [comment]
> it is good to run on multiple versions of Python to catch subtle differences between releases.

In that case maybe it's worth testing on all of 2.7, 3.6, 3.7 and 3.8 on Linux? :man_shrugging:

**Comment by cclauss** [comment]
> maybe it's worth testing on all of 2.7, 3.6, 3.7 and 3.8 on Linux?
We don't want to heat the planet too much ;-)

**Comment by lurch** [comment]
But OTOH we don't want to add features to pyftpdlib that don't work in Python3.6? (or maybe also testing on Python2.7 guarantees that'll never happen?)

File: .travis.yml [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 script:
-    - ./.ci/travis/run.sh
+    - $PYTHON setup.py install
+    - $PYTHON pyftpdlib/test/runner.py || true
+    - pytest .
+    - |
+        if [[ "$TRAVIS_OS_NAME" == 'linux' ]]; then
+            flake8 --ignore=E741,F841,W504
```
Comment (giampaolo) [inline .travis.yml]: I pushed 1122c7df99fe294be29b6093bcf538843d8e2653 so you can get rid of this

**Comment by dixonwhitmire** [comment]
Hello - is there any interest in continuing the work to resolve the test case issues? I'm happy to help, but wasn't sure if this project is still being maintained. 
Thanks!

**Comment by giampaolo** [comment]
I just merged https://github.com/giampaolo/pyftpdlib/pull/544 which replaces Travis (which is no longer free) with Github Actions, and tests pyftpdlib against python 2.7 and 3.8.

**Comment by dixonwhitmire** [comment]
@giampaolo Excellent! Thank you for the update!

**Comment by sbraz** [comment]
@giampaolo what about testing for 3.9 too? :)

**Pull request closed without merging**